pkg/reconciler/knativeserving/controller.go

```diff
@@ -39,39 +39,46 @@ import (
 // NewController initializes the controller and is called by the generated code
 // Registers eventhandlers to enqueue events
 func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
-	knativeServingInformer := knativeServinginformer.Get(ctx)
-	deploymentInformer := deploymentinformer.Get(ctx)
-	kubeClient := kubeclient.Get(ctx)
-	logger := logging.FromContext(ctx)
+	return NewExtendedController(common.NoPlatform)(ctx, cmw)
```

Comment:
If we want to extend the platform openshift, how shall we configure?
Are we creating a new configuration option to specify the platform, or changing `common.NoPlatform` into `OpenshiftPlatform`?

Reply:
A platform-specific operator would vendor this repo, then consuming it as a library. And that new operator's `sharedmain` would invoke `NewExtendedController` with its own extension.